cmd/soroban-rpc/internal/test/archive_test.go

```diff
@@ -0,0 +1,25 @@
+package test
+
+import (
+	"net/http"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestArchiveUserAgent(t *testing.T) {
+	userAgents := sync.Map{}
+	cfg := &Test{
+		historyArchiveProxyCallback: func(r *http.Request) {
+			userAgents.Store(r.Header["User-Agent"][0], "")
+		},
+	}
+	NewTest(t, cfg)
+
+	_, ok := userAgents.Load("testing")
+	assert.True(t, ok, "rpc service should set user agent for history archives")
+
+	_, ok = userAgents.Load("testing/captivecore")
```

Comment:
kinda cool to see the multiple user agents captured, there was also the captive core `curl` agent in here, but not relevant to rpc changes, so left that out.